### PR TITLE
Fix that a failed response of transient service is not logged

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
We don't log the succeeded response of a transient service.
In order to judge if it's succeeded or not, we use the status if it's server error status or not.
However, we should also take into account the response exception.

Modification:
- Check also if `responseCause` is `null` in `LoggingDecorators`.

Result:
- The failed response of a transient service is now logged properly.

### CODE REVIEW

**Review by trustin — APPROVED** [review]
:+1:

File: core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java [hunk 1]
```diff
@@ -126,7 +126,9 @@ public static void logResponse(
 
         if (responseLogLevel.isEnabled(logger)) {
             final RequestContext ctx = log.context();
-            if (!log.responseHeaders().status().isServerError() && isTransientService(ctx)) {
+            if (responseCause == null &&
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java:129]: Should we apply the same approach to request-side logs if `requestCause` is not null?

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java:129]: We currently do not log any requests of a transient service unless the option is specified.
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java#L55

Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java:129]: So IIUC, I thought we need to change the default behavior of request-side. 
Let's say a service is a `TransientService`. 
- `responseCause != null`, a response will be logged. (By this PR)
- `requestCause != null`, a request will not be logged.
  - Don't we need to log the request if a `requestCause` is not null?

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java:129]: I got it now. Yeah, that's a good idea. 😄

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox

File: core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java [hunk 2]
```diff
@@ -50,17 +49,7 @@ private LoggingDecorators() {}
...
-        if (!isTransientService) {
-            requestCompletionFuture = ctx.log().whenRequestComplete().thenApply(log -> {
-                requestLogger.accept(log);
-                return log;
-            });
-        } else {
-            requestCompletionFuture = ctx.log().whenRequestComplete();
-        }
-        requestCompletionFuture.exceptionally(e -> {
+        ctx.log().whenRequestComplete().thenAccept(requestLogger).exceptionally(e -> {
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java:52]: I feel like there are too many callbacks added here. How about just adding one callback with `handle()` and doing everything there (could catch a `Throwable`), including the `thenAccept...exceptionally` below as well?

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java]: That's a good suggestion. Fixed. :wink:

**Pull request merged**